monai/application/handlers/stats_handler.py

```diff
@@ -0,0 +1,130 @@
+import traceback
+import logging
+from ignite.engine import Engine, Events
+
+
+class StatsHandler(object):
+    """StatsHandler defines a set of Ignite Event-Handlers for all the log printing logics.
+    It's can be used for any Ignite Engine(trainer, validator and evaluator).
+    And it can support logging for epoch level and iteration level with pre-defined StatsLoggers.
+
+    Args:
+        epoch_print_logger (Callable): customized callable printer for epoch level logging.
+                                          must accept parameter "engine", use default printer if None.
+        iteration_print_logger (Callable): custimized callable printer for iteration level logging.
+                                         must accept parameter "engine", use default printer if None.
+
+    """
+    def __init__(self,
+                 epoch_print_logger=None,
+                 iteration_print_logger=None,
+                 extra_handler=None):
+        self.epoch_print_logger = epoch_print_logger
+        self.iteration_print_logger = iteration_print_logger
+        self.logger_id = 'stat_logger'
+        self.logger = logging.getLogger(logger_id)
```

Comment:
I think we should be using the engine's logger and not our own separate one. This constructor also hard-codes writing to stdout, I wouldn't put that here but let the user outside this class set where the log should go, ie. to stdout or to a file.

Reply:
if all using engine's logger, it would be difficult for the user to config output of different state handlers to different logging destination.